server/src/passport.js

```diff
@@ -0,0 +1,52 @@
+import passport from 'passport'
+import { Strategy as SteamStrategy } from 'passport-steam'
+import db from './db'
+import config from './config'
+
+passport.serializeUser((user, done) => {
+  done(null, user.id)
+})
+
+passport.deserializeUser((id, done) => {
+  db('user')
+    .first('*')
+    .where({
+      id,
+    })
+    .then(user => done(null, user))
+    .catch(err => done(err))
+})
+
+passport.use(new SteamStrategy({
+  returnURL: `${config.domain}/api/auth/callback`,
+  realm: config.domain,
+  apiKey: config.steamApiKey,
+}, (identifier, profile, done) => {
+  db('user')
+    .first('*')
+    .where({
+      steamId: profile._json.steamid, // eslint-disable-line no-underscore-dangle
+    })
+    .then((user) => {
+      if (user) {
+        return user
+      }
+      return db('user')
+        .insert({
+          steamId: profile._json.steamid, // eslint-disable-line no-underscore-dangle
+          displayName: profile._json.personaname, // eslint-disable-line no-underscore-dangle
+          profileUrl: profile._json.profileurl, // eslint-disable-line no-underscore-dangle
```

Comment:
Is all this data necessary? Could e.g. avatar urls be derived from `steamId`?

Reply:
Maybe not, but i am not sure so i think i will take the safe route and save all those for now.